### DISCUSSION TITLE
How to use canvas with dioxus-desktop

### DISCUSSION THREAD

**Discussion by n-k** [DioxusLabs/dioxus#1282] [Q&A] [answered]
Hi! 
Firstly, dioxus seems great in my limited usage so far.
The one thing I'm missing is using canvas. There is related discussion here: https://github.com/DioxusLabs/dioxus/discussions/999 but for web.

Are there any examples or guides for using canvas in a desktop app?

**Comment by ealmloff** [comment, marked as answer]
The desktop renderer doesn't support wasm-bindgen so the you cannot use web-sys.

Instead of using web-sys like the web solution, you can use the use eval function ([guide](https://dioxuslabs.com/learn/0.4/reference/desktop#running-javascript)) and [draw to the canvas with javascript](https://www.w3schools.com/html/html5_canvas.asp)
